File: Core/Entities/Product.cs
using System;

namespace Core.Entities;

public class Product : BaseEntity
{
    public string ImageUrl { get; set; } = "";
    public string Name { get; set; } = "";
    public string Description { get; set; } = "";

    //Principal
    public ICollection<ProductItem> ProductItems { get; } = new List<ProductItem>();

    //Dependent
    public int? ProductCategoryId { get; set; }
    public ProductCategory? ProductCategory { get; set; }

}

File: client/src/app/shared/models/product.ts
import { ImageUrl } from "./image";
import { Option } from "./option";

export type Product = {
    id: number;
    name: string;
    price: number;
    quantity: number;
    description: string;
    productCategoryId:number;
    productItemImgs: ImageUrl[];
    variationOpts: Option[];
}